src/test/groovy/com/jvm_bloggers/core/data_fetching/blogs/json_data/JsonBlogsDataFileValidationSpec.groovy

```diff
@@ -73,5 +65,37 @@ class JsonBlogsDataFileValidationSpec extends Specification {
         assert duplicateIds.isEmpty() : "Duplicate bookmarkable IDs found in json files: $duplicateIds"
     }
 
-}
+    @Unroll
+    def "should check that twitter user tag starts with @"() {
+        expect:
+        tag.startsWith("@")
 
+        where:
+        tag << allTwitterUsersTags()
+    }
+
+
+    @Unroll
+    def "should check that twitter user tag is less than 16 characters"() {
+        expect:
+        tag.length() <= TWITTER_USER_TAG_MAXIMUM_LENGTH
+
+        where:
+        tag << allTwitterUsersTags()
+    }
+
+    @Unroll
+    def "should check that twitter username is alphanumeric or underscore"() {
+        expect:
+        alphanumericOrUnderscore.matchesAllOf(username)
```

Comment:
When this test fails for one entry, it will be hard to say for which one. So maybe instead of `matchesAllOf` we could use @Unroll with #jsonEntry in test method name and test one entry in each run?

Reply:
Done.